src/corehost/CMakeLists.txt

```diff
@@ -1,2 +1,3 @@
 cmake_minimum_required (VERSION 2.6)
+include(../settings.cmake)
```

Comment:
This is same as that functionally. The way you are suggesting will not work because relative path of settings.cmake will be different for each project that includes setup.cmake. I would need to somehow get the absolute path of settings.cmake to make it to work.